File: Sorting/Selection Sort/selection.cpp
#include <bits/stdc++.h>
using namespace std;

void selectionSort(int arr[], int n)
{
  for (int i = 0; i < n - 1; i++)
  {
    int min = i;
    for (int j = i + 1; j < n; j++)
    {
      if (arr[j] < arr[min])
        min = j;
    }
    if (min != i)
      swap(arr[i], arr[min]);
  }
}

main()
{
  int n;
  cin >> n;
  int arr[n];
  for (int i = 0; i < n; i++)
    cin >> arr[i];

  selectionSort(arr, n);

  for (auto x : arr)
    cout << x << " ";
}

File: Sorting/Selection Sort/selection.py
def selection_sort(arr):
  n = len(arr)
  for i in range(n-1):
    min = i
    for j in range(i+1, n):
      if arr[j] < arr[min]:
        min = j
    if min != i:
      arr[min], arr[i] = arr[i], arr[min]

arr = [2,5,5,7,3,-5,6]
# [-5, 2, 3, 5, 5, 6, 7]
selection_sort(arr)
print(arr)